circuits/cpp/src/aztec3/circuits/kernel/public/native_public_kernel_circuit_private_previous_kernel.cpp

```diff
@@ -0,0 +1,62 @@
+#include "init.hpp"
+
+#include <aztec3/circuits/abis/public_kernel/public_kernel_inputs.hpp>
+#include <aztec3/circuits/abis/kernel_circuit_public_inputs.hpp>
+#include "native_public_kernel_circuit_public_previous_kernel.hpp"
+#include "common.hpp"
+
+#include <aztec3/utils/array.hpp>
+#include <aztec3/utils/dummy_composer.hpp>
+#include "aztec3/constants.hpp"
+
+namespace {
+void validate_inputs(DummyComposer& composer, PublicKernelInputs<NT> const& public_kernel_inputs)
+{
+    const auto& this_call_stack_item = public_kernel_inputs.public_call.public_call_data.call_stack_item;
+    composer.do_assert(array_length(this_call_stack_item.public_inputs.public_call_stack) > 0,
+                       "Public call stack can't be empty");
+    composer.do_assert(public_kernel_inputs.previous_kernel.public_inputs.end.private_call_count > 0,
+                       "Private call count can't be zero");
```

Comment:
We should also add a check here that the private_call_stack's length is 0. (All private functions should have been executed by now, and the stack should be empty for all public kernel iterations).